### PR TITLE
API improvements

### PR DESCRIPTION
**Pull request by spindelmanne**
Use Optional rather than returning null + improved syntax for property transfers. 

Some Javadoc and code improvements too.

### CODE REVIEW

File: readyapi4j-core/src/test/java/com/smartbear/readyapi4j/TestRecipeBuilderTest.java [hunk 1]
```diff
@@ -85,6 +89,144 @@ public void buildsRecipeWithPropertyTransferTestStep() throws Exception {
     }
 
     @Test
+    public void buildsJsonPathTransferWithImplicitSourceStep() throws Exception {
+        final String jsonPath = "$.customer.address";
+        TestRecipe recipe = newTestRecipe()
+                .addStep(getRequest("/get/something").named("theGet"))
+                .addStep(propertyTransfer(fromPreviousResponse(jsonPath)
```
Comment (olensmar) [inline readyapi4j-core/src/test/java/com/smartbear/readyapi4j/TestRecipeBuilderTest.java]: Could we distill this to `newTestRecipe( TestStepBuilder ... testSteps )` - resulting in

```
newTestRecipe( 
   getRequest(...),
   propertyTransfer( ... ),
  ...
)
```

?

File: readyapi4j-testserver-client/src/main/java/com/smartbear/readyapi4j/testserver/execution/TestServerRecipeExecutor.java [hunk 2]
```diff
@@ -38,25 +39,24 @@ public TestServerExecution submitRecipe(TestRecipe recipe) throws ApiException {
...
-        TestServerExecution execution = doExecuteTestCase(recipe.getTestCase(), recipe.getExtractorData(), true);
+        TestServerExecution execution = doExecuteTestCase(recipe.getTestCase(), true);
         notifyExecutionStarted(execution);
         return execution;
     }
 
     @Override
     public TestServerExecution executeRecipe(TestRecipe recipe) throws ApiException {
-        TestServerExecution execution = doExecuteTestCase(recipe.getTestCase(), recipe.getExtractorData(), false);
+        Optional<ExtractorData> optionoalExtractorData = recipe.getExtractorData();
```
Comment (olensmar) [inline readyapi4j-testserver-client/src/main/java/com/smartbear/readyapi4j/testserver/execution/TestServerRecipeExecutor.java]: typo...

Comment (spindelmanne) [inline readyapi4j-testserver-client/src/main/java/com/smartbear/readyapi4j/testserver/execution/TestServerRecipeExecutor.java]: yep

File: readyapi4j-core/src/test/java/com/smartbear/readyapi4j/TestRecipeBuilderTest.java (hunk 1 above)
Comment (spindelmanne) [inline readyapi4j-core/src/test/java/com/smartbear/readyapi4j/TestRecipeBuilderTest.java]: LOL, you've already added this. I'll modify the test to use the shorter syntax.

Comment (olensmar) [inline readyapi4j-core/src/test/java/com/smartbear/readyapi4j/TestRecipeBuilderTest.java]: duh... thanks :-)

File: readyapi4j-core/src/test/java/com/smartbear/readyapi4j/TestRecipeBuilderTest.java [hunk 3]
```diff
@@ -55,23 +58,20 @@ public void buildRecipeWithTestCaseProperties() {
...
-                                        .withPathLanguage(PathLanguage.XPath)
-                                )
-                                        .to(aTarget()
-                                                .withTargetStep("targetName")
-                                                .withProperty("username")
-                                                .withPath("targetPath")
-                                                .withPathLanguage(PathLanguage.XPath)
-                                        )
+        TestRecipe recipe = newTestRecipe(propertyTransfer(from(aSource()
+                        .withSourceStep("sourceName")
```
Comment (olensmar) [inline readyapi4j-core/src/test/java/com/smartbear/readyapi4j/TestRecipeBuilderTest.java]: could we condense `aSource().withSourceStep("...")` to simply `source("...")` - with an optional shortcut `previous()` ?

Comment (spindelmanne) [inline readyapi4j-core/src/test/java/com/smartbear/readyapi4j/TestRecipeBuilderTest.java]: Maybe in a second PR? Would be nice to get this one in.

**Comment by spindelmanne** [comment]
I'll fix the conflicts.

**Comment by prakashmrec** [comment]
Looks good, merging! Thanks.

**Pull request merged**